tico/passes/remove_redundant_reshape.py

```diff
@@ -295,10 +295,13 @@ def call(self, exported_program: ExportedProgram) -> PassResult:
             # Check condition
             reshape_2_input_shape = extract_shape(reshape_2_input)
             reshape_3_input_shape = extract_shape(reshape_3_input)
-            if reshape_2_input_shape != reshape_3_input_shape:
+            if not broadcastable(reshape_2_input_shape, reshape_3_input_shape):
                 continue
```

Comment:
I think this has to be true. If not, the model is somehow wrong.

```suggestion
            assert broadcastable(reshape_2_input_shape, reshape_3_input_shape)
```